rfc-164-retire-email-alert-monitoring-as-a-separate-app.md

```diff
@@ -0,0 +1,76 @@
+
+[RSS feed]: https://www.gov.uk/drug-device-alerts.atom
+
+### How email-alert-monitoring currently works for Travel Advice Alerts
+
+The app is called as a cronjob once every 15 mintues. It opens a [healthcheck URL] on the travel-advice-publisher app, which contains a list of all travel alerts published in the last 2 days, but excluding the last 150 minutes. For these, it makes a list of the subject lines of the emails that should have been sent out, as well as the time (this is because more than one travel advice alert migt be issued for the same country in a two day period, so matching purely on the subject line isn't sufficient). It then uses the GMail API to read the inbox of govuk_email_check@digital.cabinet-office.gov.uk to check for the existence of those emails. If any are missing, an alert happens. Minimum time between publishing and alert is 2 and a half hours.
+
+[healthcheck URL]: https://travel-advice-publisher.publishing.service.gov.uk/healthcheck/recently-published-editions
+
+### Problems with matching subject lines
+
+Problems can occur when trying to match by subject line if the title of the alert is altered after the email has been sent out. The [RSS feed] or [healthcheck URL] will contain a title that differs from the subject line of the email sent out. This will cause a failed match, and an alert will go off, even though for practical purposes the email has been sent. This causes false alarms, and needs someone to add a [hard-coded exception] into the matching code, increasing toil.
+
+[hard-coded execption]: https://github.com/alphagov/email-alert-monitoring/blob/main/lib/email_verifier.rb#L8-L36
+
+### What Notify API can tell us about the status of an email
+
+Notify API gives us a [current delivery status] for an email, which begins as "sending" and is changed to "delivered" when Notify receives a callback from Amazon SES (the backing service Notify uses to delivery emails) to confirm the mailserver handling emails for the recipient address has accepted the email. This is a reasonably solid assurance that the email has got to the receipient, and is definitely at the point where further handling of the email is outside of our control - the only things that might stop the recipient seeing the email at that point are:
+- The mailserver might fail to deliver internally: Valid, but we can't currently test that unless the failure occurs in the exact GMail server handling our account anyway.
+- A spam filter might redirect the message: Untestable, relies on idiosyncratic account setup, and out of our control
+- The recipient might ignore the message: Untestable, and out of our control.
+
+### How we could replace this with a job internal to email-alert-api
```

Comment:
Probably worth explicitly calling out in this section what to do about 'batched' emails (daily, weekly).  People can still sign up to travel advice / medical advice on a daily/weekly cadence - do we bother alerting on whether or not those batched emails succeeded? (I'd assume "no", since, our current monitoring doesn't do that either, and to a lesser extent because the timeliness of the email is less important).